### PR TITLE
Updated Image Details UI with additional details

### PR DESCRIPTION
**Pull request by dabasnidhi**
For Testing - http://nidabas-desk/DefaultCollection/p1/_serviceEndpoints/09b3aa23-0c50-4958-9d59-5f5e42ec364d/kubernetesSummary?namespace=kube-system

image

### CODE REVIEW

**Comment by madhuv-msft** [comment]
> For Testing - http://nidabas-desk/DefaultCollection/p1/_serviceEndpoints/09b3aa23-0c50-4958-9d59-5f5e42ec364d/kubernetesSummary?namespace=kube-system
> 
> image

please make sure Labels are oneline or two. we have something called Tags, please use it.
also date and time, should we go with short date or long date?

File: src/Contracts/Types.d.ts [hunk 1]
```diff
@@ -13,14 +13,17 @@ export interface IImageDetails {
...
-    buildId: number;
-    buildVersion: string;
-    buildDefinitionName: string;
-    buildDefinitionId: string;
+    runId: number;
+    pipelineVersion: string;
+    pipelineName: string;
+    pipelineId: string;
+    jobName: string;
+    imageSize: string;
```
Comment (madhuv-msft) [inline src/Contracts/Types.d.ts]: as discussed we need to have URL also for the link purpose.

File: src/WebUI/ImageDetails/ImageDetails.tsx [hunk 2]
```diff
@@ -48,6 +48,7 @@ export class ImageDetails extends BaseComponent<IImageDetailsProperties, IImageD
     private _getMainHeading(): JSX.Element | null {
         const imageDetails = this.props.imageDetails;
         this._displayImageName = Utils.extractDisplayImageName(imageDetails.imageName);
+        const jobName: string = format("{0} #{1} on {2}", imageDetails.jobName || "", imageDetails.pipelineVersion || "", imageDetails.pipelineName || "");
```
Comment (madhuv-msft) [inline src/WebUI/ImageDetails/ImageDetails.tsx]: this seems to be a link, and also formatlocale, this has to be localized.

File: src/WebUI/ImageDetails/ImageDetails.tsx [hunk 3]
```diff
@@ -148,6 +152,7 @@ export class ImageDetails extends BaseComponent<IImageDetailsProperties, IImageD
         const contentClassName = "image-o-v-col-content";
         switch (key) {
             case Resources.LabelsText:
+            case Resources.TagsText:
```
Comment (madhuv-msft) [inline src/WebUI/ImageDetails/ImageDetails.tsx]: Use Tags component, do not use LabelGroup

File: src/WebUI/ImageDetails/ImageDetails.tsx [hunk 4]
```diff
@@ -261,14 +266,15 @@ export class ImageDetails extends BaseComponent<IImageDetailsProperties, IImageD
...
+        const textToRender = imageLayer.size || "-";
         const itemToRender = defaultColumnRenderer(textToRender);
         return renderTableCell(rowIndex, columnIndex, tableColumn, itemToRender);
     }
 
     private static _renderLayersAgeCell(rowIndex: number, columnIndex: number, tableColumn: ITableColumn<IImageLayer>, imageLayer: IImageLayer): JSX.Element {
         // Currently created data is not present in imageLayer
-        const itemToRender = <Ago date={new Date()} format={AgoFormat.Extended} />;
+        const layerCreatedOn = imageLayer.createdOn ? new Date(imageLayer.createdOn) : new Date();
+        const itemToRender = <Ago date={layerCreatedOn} format={AgoFormat.Extended} />;
```
Comment (madhuv-msft) [inline src/WebUI/ImageDetails/ImageDetails.tsx]: check with PM on extented or normal format

File: src/WebUI/ImageDetails/ImageDetails.tsx (hunk 2 above)
Comment (madhuv-msft) [inline src/WebUI/ImageDetails/ImageDetails.tsx]: please note if we do not have any of these details, we should not show "# on " string, we should skip the subtext altogether.

File: src/WebUI/ImageDetails/ImageDetails.tsx [hunk 5]
```diff
@@ -61,6 +62,9 @@ export class ImageDetails extends BaseComponent<IImageDetailsProperties, IImageD
                             {this._displayImageName}
                         </HeaderTitle>
                     </HeaderTitleRow>
+                    <HeaderDescription className={"text-ellipsis"}>
+                        {jobName}
+                    </HeaderDescription>
```
Comment (madhuv-msft) [inline src/WebUI/ImageDetails/ImageDetails.tsx]: singleClickActivation is false, still on hover 'hand' is shown for cursor. we should check why this is happening.

Comment (madhuv-msft) [inline src/WebUI/ImageDetails/ImageDetails.tsx]: we should addp padding at the end of the image details, not in the layers, but in the "Image details" section, please pod overview page.

Comment (dabasnidhi) [reply src/WebUI/ImageDetails/ImageDetails.tsx]: Top and bottom padding of 5px is added via class image-o-v-col-content..Its the same in Pod Overview

File: src/Contracts/Types.d.ts (hunk 1 above)
Comment (dabasnidhi) [reply src/Contracts/Types.d.ts]: Added changes to extract url from labels and using that in RunDetails

File: src/WebUI/ImageDetails/ImageDetails.tsx (hunk 2 above)
Comment (dabasnidhi) [reply src/WebUI/ImageDetails/ImageDetails.tsx]: Added changes to extract url from labels and using that in RunDetails

File: src/WebUI/ImageDetails/ImageDetails.tsx (hunk 4 above)
Comment (dabasnidhi) [reply src/WebUI/ImageDetails/ImageDetails.tsx]: Changed to Compact format as per spec

File: src/WebUI/ImageDetails/ImageDetails.tsx (hunk 2 above)
Comment (dabasnidhi) [reply src/WebUI/ImageDetails/ImageDetails.tsx]: Added changes to extract url from labels and using that in RunDetails

**Pull request merged**